Resources/public/js/views/ez-languageselectionboxview.js

```diff
@@ -0,0 +1,412 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-languageselectionboxview', function (Y) {
+    "use strict";
+    /**
+     * Provides the language selection box view class
+     *
+     * @method ez-languageselectionboxview
+     */
+    Y.namespace('eZ');
+
+    var CONFIRM = 'confirm',
+        CANCEL = 'cancel',
+        BASE_TRANSLATION_AVAILABLE_CLASS = 'is-base-translation-allowed',
+        BASE_TRANSLATIONS_LIST_VISIBLE_CLASS = 'is-base-translations-list-visible',
+        TRANSLATION_HIGHLIGHT_CLASS = 'is-translation-selected';
+
+    /**
+     * The language selection box view.
+     *
+     * @namespace eZ
+     * @class LanguageSelectionBoxView
+     * @constructor
+     * @extends eZ.TemplateBasedView
+     */
+    Y.eZ.LanguageSelectionBoxView = Y.Base.create('languageSelectionBoxView', Y.eZ.TemplateBasedView, [], {
+        events: {
+            '.ez-languageselectionbox-close': {
+                'tap': '_cancel',
+            },
+            '.ez-languageselectionbox-confirm': {
+                'tap': '_confirm',
+            },
+            '.ez-new-translation': {
+                'tap': '_selectNewTranslation'
+            },
+            '.ez-base-translation': {
+                'tap': '_selectBaseTranslation'
+            },
+            '.ez-base-translation-checkbox': {
+                'change': '_toggleBaseTranslation'
+            },
+        },
+
+        initializer: function () {
+            this.on(['confirmHandlerChange', 'cancelHandlerChange'], function (e) {
+                this._syncEventHandler(e.attrName.replace(/Handler$/, ''), e.prevVal, e.newVal);
+            });
+            this._syncEventHandler('confirm');
+            this._syncEventHandler('cancel');
+            this.after('activeChange', function () {
+                if ( this.get('active') ) {
+                    this._uiUpdateTitle();
+                }
+            });
+            this._publishEvents();
+            this.on('existingTranslationsChange', function (e) {
```

Comment:
It seems that this part is not working as I wanted... Event is triggered but if call `this.render()` here it crashed inside method `_getNewTranslations` which is getter for `newTranslations` attribute. It seems that at the time when event is triggered, the value of attribute `availableTranslations` from `LanguageSelectionBoxViewService` is not set and `this.get('availableTranslations')` returns `undefined`.